src/Shop/ExternalCalculatorService/ElixirPort.php

```diff
@@ -0,0 +1,28 @@
+<?php
+namespace DDDHH\Shop\ExternalCalculatorService;
+
+use DDDHH\Hexagon\Port;
+use GuzzleHttp\Client;
+use GuzzleHttp\Exception\RequestException;
+use GuzzleHttp\Psr7;
+
+class ElixirPort extends Port {
```

Comment:
This abstraction is shaky. Since you extend `Hexagon` which in turn allows **any** `Port` implementation it's not safe to do so.

You could of course mitigate those inheritance problems by introducing a `PortResponse` interface.

Anyway, for me the Hexagon is a way of designing systems and it doesn't have to be made explicit this way. It's way too generic.